modules/vulnix.nix

```diff
@@ -0,0 +1,174 @@
+        or <command>vulnix --help</command> for more information.
+      '';
+    };
+
+    sink = mkOption {
+      type = types.path;
+      description = ''
+        Program that processes the result of each scan. It receives the vulnix output on stdin.
+        When receiving the result of nomad job scans the environment variables
+        <envar>NOMAD_JOB_NAMESPACE</envar>, <envar>NOMAD_JOB_ID</envar>,
+        <envar>NOMAD_JOB_TASKGROUP_NAME</envar>, and <envar>NOMAD_JOB_TASK_NAME</envar> are set.
+      '';
+    };
+
+    sshKey = mkOption {
+      type = types.path;
+      description = "The SSH key to use for private Git repos.";
+    };
+
+    netrcFile = mkOption {
+      type = types.path;
+      description = "The netrc file to use for private Git repos.";
+    };
+  };
+
+  config.systemd = lib.mkIf cfg.enable {
+    services.vulnix = {
+      description = "Vulnix scan";
+
+      serviceConfig = {
+        Type = "oneshot";
+        DynamicUser = true;
+        CacheDirectory = "vulnix";
+        StateDirectory = "vulnix";
+      } // lib.optionalAttrs cfg.scanNomadJobs {
+        Type = "simple";
+        Restart = "on-failure";
+        LoadCredential = [
+          (assert config.services.vault-agent-core.enable; "vault-token:/run/keys/vault-token")
+          "ssh:${cfg.sshKey}"
+          "netrc:${cfg.netrcFile}"
+        ];
+      };
+
+      startLimitIntervalSec = 20;
+      startLimitBurst = 10;
+
+      environment = lib.mkIf cfg.scanNomadJobs {
+        VAULT_ADDR = "https://${config.cluster.instances.core-1.privateIP}:8200";
+        NOMAD_ADDR = "https://${config.cluster.instances.core-1.privateIP}:4646";
+        VAULT_CACERT = "/etc/ssl/certs/full.pem";
+      };
+
+      path = with pkgs; [ cfg.package vault-bin curl jq nixFlakes gitMinimal ];
+
+      script = ''
+        set -o pipefail
+
+        function scan {
+          vulnix ${lib.cli.toGNUCommandLineShell {} (with cfg; {
+            json = true;
+            requisites = scanRequisites;
+            no-requisites = !scanRequisites;
+          })} \
+            --cache-dir $CACHE_DIRECTORY \
+            ${lib.concatStringsSep " " cfg.extraOpts} "$@" \
+          || case $? in
+            0 ) ;; # no vulnerabilities found
+            1 ) ;; # only whitelisted vulnerabilities found
+            2 ) ;; # vulnerabilities found
+            * ) exit $? ;; # unexpected
+          esac
+        }
+
+        scan ${lib.cli.toGNUCommandLineShell {} (with cfg; {
+          system = scanSystem;
+          gc-roots = scanGcRoots;
+        })} \
+          -- ${lib.escapeShellArgs cfg.paths} \
+        | ${cfg.sink}
+      '' + lib.optionalString cfg.scanNomadJobs ''
+        export VAULT_TOKEN=$(< $CREDENTIALS_DIRECTORY/vault-token)
+        NOMAD_TOKEN=$(vault read -format json -field secret_id nomad/creds/admin | jq -rj)
+        sleep 5s # let nomad token be propagated to come into effect
```

Comment:
Probably way too long. Could test the token in a loop until it works. Time is not critical here though...